nemo/collections/nlp/models/language_modeling/transformer_lm_model.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import os
+from typing import Dict, Optional
+
+import torch
+from omegaconf import DictConfig
+from pytorch_lightning import Trainer
+
+from nemo.collections.common.losses import SmoothedCrossEntropyLoss
+from nemo.collections.common.tokenizers import CharTokenizer, WordTokenizer
+from nemo.collections.common.tokenizers.tokenizer_utils import get_tokenizer
+from nemo.collections.nlp.data import L2RLanguageModelingDataset
+from nemo.collections.nlp.modules.common import TokenClassifier
+from nemo.collections.nlp.modules.common.transformer import (
+    TransformerEmbedding,
+    TransformerEncoder,
+    transformer_weights_init,
+)
+from nemo.core.classes.common import typecheck
+from nemo.core.classes.modelPT import ModelPT
+from nemo.utils.decorators import experimental
+
+__all__ = ['TransformerLMModel']
+
+
+@experimental
+class TransformerLMModel(ModelPT):
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        """Initializes the BERTTextClassifier model.
+        """
+
+        # shared params for dataset and data loaders
+        self.dataset_cfg = cfg.dataset
+
+        self.tokenizer = globals()[cfg.language_model.tokenizer](vocab_file=cfg.language_model.vocab_file)
```

Comment:
could you use get_tokenizer to be consistent with other nlp models  i.e. https://github.com/NVIDIA/NeMo/blob/candidate/nemo/collections/nlp/models/language_modeling/lm_model.py#L233

Reply:
^^